strelets/adapter/swarm_prepare_reverse_proxy.go

```diff
@@ -13,7 +13,8 @@ type ReverseProxyConfig struct {
 }
 
 func (d *dockerSwarm) PrepareReverseProxy(ctx context.Context, config *ReverseProxyConfig) (Runner, error) {
-	if err := d.RemoveContainer(ctx, PROXY_CONTAINER_NAME); err != nil {
+	serviceName := GetServiceId(PROXY_CONTAINER_NAME)
```

Comment:
@netoneko what was the bug here, and where's the test that reproduces it?

Reply:
The bug was that the service name passed was not suffixed with `-stack`, therefore when docker looked up `chain-1000` to disable it, it also found `chain-1000001` and it was disabled too.